### PR TITLE
add admin role if user is admin

### PR DESCRIPTION
**Pull request by jakepanter**
sorry for the formatting stuff, did not intend that, but we can have some common formatting rules in api later, I guess.

- on user creation we check if the user is in the list of admins (comma-seperated list from secrets.json)
- always add Participant role

### CODE REVIEW

**Review by LetsPoke — APPROVED** [review]

File: hms-api/src/handler/user/create.ts [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-import {buildResponse} from '../../rest/responses';
-import {createUser} from '../../service/user-service';
-import {wrapHandler} from '../handler-wrapper';
-import UserCreateRequest from '../../rest/UserCreateRequest';
-import UserCreateResponse from '../../rest/UserCreateResponse';
+import { buildResponse } from "../../rest/responses";
+import { createUser } from "../../service/user-service";
+import { wrapHandler } from "../handler-wrapper";
+import UserCreateRequest from "../../rest/UserCreateRequest";
+import UserCreateResponse from "../../rest/UserCreateResponse";
```
Comment (verzano) [inline hms-api/src/handler/user/create.ts:7]: single quotes is right for the backend, and no space between the import and the `{` or `}`

File: hms-api/src/handler/user/create.ts [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-        request.firstName,
-        request.emailAddress,
-        request.roles,
-        request.skills,
-        request.imageUrl,
+      request.lastName,
+      request.firstName,
+      request.emailAddress,
+      request.skills,
+      request.imageUrl
```
Comment (verzano) [inline hms-api/src/handler/user/create.ts:17]:
```suggestion
      request.imageUrl,
```

File: hms-api/src/handler/user/create.ts (hunk 1 above)
Comment (verzano) [reply hms-api/src/handler/user/create.ts:7]: goes for all the other files too

File: hms-api/src/handler/user/create.ts (hunk 2 above)
Comment (verzano) [reply hms-api/src/handler/user/create.ts:17]: multiline lists of arguments should have an `.` on every line

File: hms-api/src/service/user-service.ts [hunk 3]
```diff
@@ -117,9 +129,10 @@ export function extractUser(users: User[], participant: Participant): User {
 
 async function verifyAllSkillsExist(skillIds: Uuid[]): Promise<void> {
   for (const skillId of skillIds) {
-    if (!await skillExists(skillId)) {
-      throw new ReferenceNotFoundError(`Cannot create Idea, ` +
-          `Skill with id: ${skillId} does not exist`);
+    if (!(await skillExists(skillId))) {
+      throw new ReferenceNotFoundError(
+        `Cannot create Idea, ` + `Skill with id: ${skillId} does not exist`
```
Comment (verzano) [inline hms-api/src/service/user-service.ts:134]: I mean if this is all going to be on line the `+` is not necessary

File: hms-api/src/repository/category-repository.ts [hunk 4]
```diff
@@ -59,39 +66,46 @@ export async function getCategory(id: Uuid): Promise<Category> {
...
+  );
 
   if (output.Attributes) {
     return itemToCategory(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete Category with id: ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete Category with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/category-repository.ts:100]:
```suggestion
    `Cannot delete Category with id: ${id}, it does not exist`,
```

File: hms-api/src/repository/domain/Hackathon.ts [hunk 5]
```diff
@@ -81,34 +81,35 @@ class Hackathon {
     }
 
     if (this.startDate >= this.endDate) {
-      result.addFailure(`startDate (${this.startDate}) ` +
-          `is after endDate (${this.endDate})`);
+      result.addFailure(
+        `startDate (${this.startDate}) ` + `is after endDate (${this.endDate})`,
```
Comment (verzano) [inline hms-api/src/repository/domain/Hackathon.ts:85]:
```suggestion
        `startDate (${this.startDate}) is after endDate (${this.endDate})`,
```

File: hms-api/src/repository/hackathon-repository.ts [hunk 6]
```diff
@@ -58,36 +64,41 @@ export async function getHackathon(id: Uuid): Promise<Hackathon> {
...
+  );
 
   if (output.Attributes) {
     return itemToHackathon(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete Hackathon with id: ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete Hackathon with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/hackathon-repository.ts:91]:
```suggestion
    `Cannot delete Hackathon with id: ${id}, it does not exist`,
```

File: hms-api/src/repository/idea-repository.ts [hunk 7]
```diff
@@ -130,60 +141,67 @@ export async function getIdea(id: Uuid): Promise<Idea> {
...
+  );
 
   if (output.Attributes) {
     return itemToIdea(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete Idea with id ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete Idea with id ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/idea-repository.ts:157]:
```suggestion
    `Cannot delete Idea with id ${id}, it does not exist`,
```

File: hms-api/src/repository/participant-repository.ts [hunk 8]
```diff
@@ -76,58 +85,65 @@ export async function getParticipants(ids: Uuid[]): Promise<Participant[]> {
...
+  );
 
   if (output.Attributes) {
     return itemToParticipant(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete Participant with id: ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete Participant with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/participant-repository.ts:138]:
```suggestion
    `Cannot delete Participant with id: ${id}, it does not exist`,
```

File: hms-api/src/repository/skill-repository.ts [hunk 9]
```diff
@@ -63,33 +69,34 @@ export async function getSkills(ids: Uuid[]): Promise<Skill[]> {
...
+  );
 
   if (output.Attributes) {
     return itemToSkill(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete Skill with id: ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete Skill with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/skill-repository.ts:96]:
```suggestion
    `Cannot delete Skill with id: ${id}, it does not exist`,
```

File: hms-api/src/repository/user-repository.ts [hunk 10]
```diff
@@ -73,52 +79,59 @@ export async function getUsers(ids: Uuid[]): Promise<User[]> {
...
+  );
 
   if (output.Attributes) {
     return itemToUser(output.Attributes);
   }
 
-  throw new NotFoundError(`Cannot delete User with id: ${id}, ` +
-      `it does not exist`);
+  throw new NotFoundError(
+    `Cannot delete User with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/repository/user-repository.ts:122]: What did all this autoformatting? I am not a fan :(
```suggestion
    `Cannot delete User with id: ${id}, it does not exist`,
```

File: hms-api/src/service/category-service.ts [hunk 11]
```diff
@@ -42,18 +44,19 @@ export async function createCategory(
...
   let existing: Category;
   try {
     existing = await getCategory(id);
     existing.title = title;
     existing.description = description;
   } catch (e) {
-    throw new NotFoundError(`Cannot edit Category with id: ${id}, ` +
-        `it does not exist`);
+    throw new NotFoundError(
+      `Cannot edit Category with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/service/category-service.ts:58]:
```suggestion
      `Cannot edit Category with id: ${id}, it does not exist`,
```

File: hms-api/src/service/hackathon-service.ts [hunk 12]
```diff
@@ -41,63 +41,67 @@ export async function createHackathon(
...
+  id: Uuid,
 ): Promise<HackathonResponse> {
   let participants;
   try {
     participants = await listParticipants(id);
   } catch (e) {
-    throw new ReferenceNotFoundError(`Cannot get Hackathon with id: ${id}, ` +
-        `unable to list Participants`);
+    throw new ReferenceNotFoundError(
+      `Cannot get Hackathon with id: ${id}, ` + `unable to list Participants`,
```
Comment (verzano) [inline hms-api/src/service/hackathon-service.ts:51]:
```suggestion
      `Cannot get Hackathon with id: ${id}, unable to list Participants`,
```

File: hms-api/src/service/hackathon-service.ts [hunk 13]
```diff
@@ -41,63 +41,67 @@ export async function createHackathon(
...
   }
 
   let categories;
   try {
     categories = await listCategories(id);
   } catch (e) {
-    throw new ReferenceNotFoundError(`Cannot get Hackathon with id: ${id}, ` +
-        `unable to list Categories`);
+    throw new ReferenceNotFoundError(
+      `Cannot get Hackathon with id: ${id}, ` + `unable to list Categories`,
```
Comment (verzano) [inline hms-api/src/service/hackathon-service.ts:71]:
```suggestion
      `Cannot get Hackathon with id: ${id}, unable to list Categories`,
```

File: hms-api/src/service/hackathon-service.ts [hunk 14]
```diff
@@ -41,63 +41,67 @@ export async function createHackathon(
...
   }
 
   let ideas;
   try {
     ideas = await listIdeasForHackathon(id);
   } catch (e) {
-    throw new ReferenceNotFoundError(`Cannot get Hackathon with id: ${id}, ` +
-        `unable to list Ideas`);
+    throw new ReferenceNotFoundError(
+      `Cannot get Hackathon with id: ${id}, ` + `unable to list Ideas`,
```
Comment (verzano) [inline hms-api/src/service/hackathon-service.ts:80]:
```suggestion
      `Cannot get Hackathon with id: ${id}, unable to list Ideas`,
```

File: hms-api/src/service/hackathon-service.ts [hunk 15]
```diff
@@ -107,8 +111,9 @@ export async function editHackathon(
     existing.startDate = startDate;
     existing.endDate = endDate;
   } catch (e) {
-    throw new NotFoundError(`Cannot edit Hackathon with id: ${id}, ` +
-        `it does not exist`);
+    throw new NotFoundError(
+      `Cannot edit Hackathon with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/service/hackathon-service.ts:115]:
```suggestion
      `Cannot edit Hackathon with id: ${id}, it does not exist`,
```

File: hms-api/src/service/idea-service.ts [hunk 16]
```diff
@@ -36,44 +36,51 @@ import InvalidStateError from '../error/InvalidStateError';
...
+    );
   }
 
   for (const skillId of requiredSkills) {
-    if (!await skillExists(skillId)) {
-      throw new ReferenceNotFoundError(`Cannot create Idea, ` +
-          `Skill with id: ${skillId} does not exist`);
+    if (!(await skillExists(skillId))) {
+      throw new ReferenceNotFoundError(
+        `Cannot create Idea, ` + `Skill with id: ${skillId} does not exist`,
```
Comment (verzano) [inline hms-api/src/service/idea-service.ts:70]:
```suggestion
        `Cannot create Idea, Skill with id: ${skillId} does not exist`,
```

File: hms-api/src/service/idea-service.ts [hunk 17]
```diff
@@ -85,32 +92,37 @@ export async function createIdea(
...
+  categoryId: Uuid,
 ): Promise<void> {
   let existing: Idea;
   try {
     existing = await getIdea(id);
   } catch (e) {
-    throw new NotFoundError(`Cannot edit Idea with id: ${id}, ` +
-        `it does not exist`);
+    throw new NotFoundError(
+      `Cannot edit Idea with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/service/idea-service.ts:108]:
```suggestion
      `Cannot edit Idea with id: ${id}, it does not exist`,
```

File: hms-api/src/service/participant-service.ts [hunk 18]
```diff
@@ -25,15 +25,18 @@ import NotFoundError from '../error/NotFoundError';
...
-    throw new ReferenceNotFoundError(`Cannot create Participant, ` +
-        `User with id: ${userId} does not exist`);
+  if (!(await hackathonExists(hackathonId))) {
+    throw new ReferenceNotFoundError(
+      `Cannot create Participant, ` +
+        `Hackathon with id: ${hackathonId} does not exist`,
+    );
+  } else if (!(await userExists(userId))) {
+    throw new ReferenceNotFoundError(
+      `Cannot create Participant, ` + `User with id: ${userId} does not exist`,
```
Comment (verzano) [inline hms-api/src/service/participant-service.ts:38]:
```suggestion
      `Cannot create Participant, User with id: ${userId} does not exist`,
```

File: hms-api/src/service/skill-service.ts [hunk 19]
```diff
@@ -29,18 +29,19 @@ export async function createSkill(
...
   let existing;
   try {
     existing = await getSkill(id);
     existing.name = name;
     existing.description = description;
   } catch (e) {
-    throw new NotFoundError(`Cannot edit Skill with id: ${id}, ` +
-        `it does not exist`);
+    throw new NotFoundError(
+      `Cannot edit Skill with id: ${id}, ` + `it does not exist`,
```
Comment (verzano) [inline hms-api/src/service/skill-service.ts:43]:
```suggestion
      `Cannot edit Skill with id: ${id}, it does not exist`,
```

File: hms-api/src/service/user-service.ts [hunk 20]
```diff
@@ -90,8 +101,9 @@ export async function editUser(
     existing.skills = skills;
     existing.imageUrl = imageUrl;
   } catch (e) {
-    throw new NotFoundError(`Cannot edit User with id: ${id}, ` +
-        `it does not exist`);
+    throw new NotFoundError(
+      `Cannot edit User with id: ${id}, ` + 'it does not exist',
```
Comment (verzano) [inline hms-api/src/service/user-service.ts:105]:
```suggestion
      `Cannot edit User with id: ${id}, it does not exist`,
```

**Pull request merged**